SpatialGDK/Source/SpatialGDKFunctionalTests/Private/GenerateTestMapsCommandlet.cpp

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "GenerateTestMapsCommandlet.h"
+#include "TestMaps/GeneratedTestMap.h"
+
+DEFINE_LOG_CATEGORY(LogGenerateTestMapsCommandlet);
+
+UGenerateTestMapsCommandlet::UGenerateTestMapsCommandlet()
+{
+	IsClient = false;
+	IsEditor = true;
+	IsServer = false;
+	LogToConsole = true;
+}
+
+int32 UGenerateTestMapsCommandlet::Main(const FString& CmdLineParams)
+{
+	UE_LOG(LogGenerateTestMapsCommandlet, Display, TEXT("Generate test maps commandlet started."));
+
+	UE_LOG(LogGenerateTestMapsCommandlet, Display, TEXT("Deleting the %s folder."), *UGeneratedTestMap::GetGeneratedMapFolder());
+	if (FPaths::DirectoryExists(UGeneratedTestMap::GetGeneratedMapFolder()))
+	{
+		IPlatformFile& PlatformFile = FPlatformFileManager::Get().GetPlatformFile();
+		PlatformFile.DeleteDirectoryRecursively(*UGeneratedTestMap::GetGeneratedMapFolder());
+	}
+
+	// Have to gather the classes first and then iterate over the copy, because creating a map triggers a GC which can modify the object
+	// array this is iterating through
+	TArray<UClass*> TestMapClasses;
+	for (TObjectIterator<UClass> Iter; Iter; ++Iter)
+	{
+		if (Iter->IsChildOf(UGeneratedTestMap::StaticClass()) && *Iter != UGeneratedTestMap::StaticClass())
+		{
+			TestMapClasses.Add(*Iter);
+		}
+	}
+
+	for (UClass* TestMapClass : TestMapClasses)
+	{
+		UGeneratedTestMap* TestMap = NewObject<UGeneratedTestMap>(this, TestMapClass);
+		TestMap->AddToRoot(); // Okay, must admit, not completely sure what's going on here, seems like even though the commandlet is
```

Comment:
I think you can add it to a UPROPERTY() on the commandlet so this object is referenced and isn't GC'd, just having an outer is insufficient.

Reply:
Valentyn also mentioned that maybe the commandlet itself gets gced, and we wouldn't know since we do not access this after that point.